package test

trait A
trait B : A, <!CYCLIC_INHERITANCE_HIERARCHY!>E<!>
trait C : <!CYCLIC_INHERITANCE_HIERARCHY!>B<!>
trait D : <!CYCLIC_INHERITANCE_HIERARCHY!>B<!>
trait E : <!CYCLIC_INHERITANCE_HIERARCHY!>F<!>
trait F : <!CYCLIC_INHERITANCE_HIERARCHY!>D<!>, <!CYCLIC_INHERITANCE_HIERARCHY!>C<!>
trait G : F
trait H : F